source/generators/SingleParticle.cc

```diff
@@ -35,8 +35,8 @@ using namespace CLHEP;
 
 SingleParticle::SingleParticle():
 G4VPrimaryGenerator(), msg_(0), particle_definition_(0),
-energy_min_(0.), energy_max_(0.), geom_(0), momentum_X_(0.),
-momentum_Y_(0.), momentum_Z_(0.), costheta_min_(-1.), costheta_max_(1.),
+energy_min_(0.), energy_max_(0.), geom_(0),
+costheta_min_(-1.), costheta_max_(1.),
```

Comment:
You should initialize `momentum_` here: Just add `momentum_{}`, or `momentum_{0,0,0}` if you want to be more explicit. Alternatively add `{}` or `{0,0,0}` at the point where you declare the `momentum_` member in the header.

Reply:
Done